pandas/core/generic.py

```diff
@@ -5557,12 +5557,17 @@ def astype(
                     results.append(col.copy() if copy else col)
 
         elif is_extension_array_dtype(dtype) and self.ndim > 1:
-            # GH 18099/22869: columnwise conversion to extension dtype
-            # GH 24704: use iloc to handle duplicate column names
-            results = [
-                self.iloc[:, i].astype(dtype, copy=copy)
-                for i in range(len(self.columns))
-            ]
+            # GH 33113: empty dataframe as extension dtype, concat ValueError
+            if len(self.columns) > 0:
```

Comment:
we do not use specific checks like this
ether the issue is deeper in astype

Reply:
what's happening here is that if the dataframe has no column, `results == []` and then `pd.concat(results, ...)` fails because `pd.concat([])` raises a `ValueError`

Reply:
what would be a "non-specific" check here? not sure about what you mean by "specific check"